iree/hal/vmla/op_kernels_ruy.h

```diff
@@ -18,8 +18,8 @@
 #include "absl/base/thread_annotations.h"
 #include "absl/memory/memory.h"
 #include "iree/base/status.h"
-#include "tensorflow/lite/experimental/ruy/context.h"
-#include "tensorflow/lite/experimental/ruy/ruy.h"
+#include "third_party/ruy/context.h"
+#include "third_party/ruy/ruy.h"
```

Comment:
I think we should rather add `${CMAKE_CURRENT_SOURCE_DIR}/third_party/` to `IREE_COMMON_INCLUDE_DIRS` in [iree_copts.cmake](https://github.com/google/iree/blob/master/build_tools/cmake/iree_copts.cmake) instead of changing the include here?

Reply:
Are you proposing having the include path as 'ruy/context.h' instead of 'third_party/ruy/context.h' in IREE (and the same for other libraries)? I think we should evaluate that separately from the ruy move.

In the upstream repository (where this change is originating), 'third_party/some_library_name/directory/file.h' is the default include format. We can diverge from that either only on GitHub or in both repositories (with Copybara rewrites, of which we already have a few), there's just a bit of effort needed. IIRC, the header include format is just a detail of the library, and whichever decision we make in IREE won't affect projects using IREE.

With Dear ImGui, I added the third_party/ paths back like this, for consistency:

https://github.com/google/iree/blob/c78eee264dd3f20d123997e6881ed974af5842f3/build_tools/third_party/dear_imgui/BUILD.overlay#L17-L27

https://github.com/google/iree/blob/c78eee264dd3f20d123997e6881ed974af5842f3/build_tools/third_party/dear_imgui/CMakeLists.txt#L15-L36

Reply:
Just realized that the change would match with the way headers from dear_imgui and dawn are included. TensorFlow is also using the include paths that way.

However, one could argue is that this probably prohibits the use of libraries installed to your system...

Reply:
:+1: To integrate this as is.

Reply:
Thanks for raising the important issue of ruy CMake optimal integration. Yes eventually it could be interesting to have CMake support upstreamed in ruy. Thanks for being OK to handle this separately from the move. The move is proving extremely hard to do by itself so I can't widen the scope to include CMake considerations right now.

Reply:
Feel free to open an issue in the ruy repository and to mention me :)